src/core/services/geolocation.ts

```diff
@@ -133,6 +135,30 @@ export class CoreGeolocationProvider {
         return CoreUtils.promiseWorks(Diagnostic.getLocationAuthorizationStatus());
     }
 
+    /**
+     * Request and return the location authorization status for the application.
+     *
+     * @return Promise resolved with authorization status.
+     */
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    protected requestLocationAuthorization(): Promise<any> {
+        if (!CoreApp.isIOS()) {
+            return Diagnostic.requestLocationAuthorization();
+        }
+
+        // In iOS, the modal disappears when the screen is locked and the promise never ends. Treat that case.
+        return new Promise((resolve, reject) => {
+            const pauseSubscription = CorePlatform.pause.subscribe(() => {
+                pauseSubscription?.unsubscribe();
+                reject(new CoreSilentError()); // Don't display an error, just finish the process.
+            });
+
+            Diagnostic.requestLocationAuthorization().then(resolve, reject).finally(() => {
+                pauseSubscription.unsubscribe();
+            });
```

Comment:
```suggestion
            // Don't display an error, just finish the process.
            const unsubscribe = CoreSubscriptions.once(CorePlatform.pause, () => reject(new CoreSilentError()));

            Diagnostic.requestLocationAuthorization().then(resolve, reject).finally(() => unsubscribe());
```

I recently found a similar use-case where I wanted to use `CoreSubscriptions.once`, but couldn't because the subscription could be cancelled elsewhere. What do you think of returning an unsubscription callback as well, to improve this type of scenario?

Reply:
Sounds good, done!